linux/hid.c

```diff
@@ -1326,6 +1383,9 @@ HID_API_EXPORT struct hid_device_info *HID_API_CALL hid_get_device_info(hid_devi
 
 int HID_API_EXPORT_CALL hid_get_indexed_string(hid_device *dev, int string_index, wchar_t *string, size_t maxlen)
 {
+	if (!dev)
+		return -1;
```

Comment:
all public API (specifically functions that accept `hid_device *` as first argument):
- do not make sens to call with non-valid `dev`
- should register an error string in case if anything goes wrong, so the user could use `hid_error` to check what went wrong in case if `-1` is returned

I do not support having this checks for all public API functions (marked with `HID_API_CALL`/`HID_API_EXPORT_CALL`).
We always assume that the `dev` should be valid, and that is up to the user to make sure it is. (It is fairly easy to implement in other languages that wrap this library w/o any additional runtime checks by using e.g. class invariant).
The only exception is `hid_close` function - similar to `free` it should gracefully accept NULL-device.

All other functions the accept `hid_device *` as an argument called from public API function should use the same assumption, that the `dev` is valid and doesn't require extra checks.